pkg/recovery/apiserver.go

```diff
@@ -0,0 +1,438 @@
+package recovery
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path"
+	"text/template"
+	"time"
+
+	"github.com/ghodss/yaml"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	clientcmdapiv1 "k8s.io/client-go/tools/clientcmd/api/v1"
+	"k8s.io/klog"
+
+	"github.com/openshift/library-go/pkg/crypto"
+
+	"github.com/openshift/cluster-kube-apiserver-operator/pkg/operator/v311_00_assets"
+)
+
+const (
+	KubeApiserverStaticPodFileName = "kube-apiserver-pod.yaml"
+	RecoveryPodFileName            = "recovery-kube-apiserver-pod.yaml"
+	RecoveryPodNamespace           = "kube-system"
+	RecoveryCofigFileName          = "config.yaml"
+	AdminKubeconfigFileName        = "admin.kubeconfig"
+
+	RecoveryPodAsset    = "v3.11.0/kube-apiserver/recovery-pod.yaml"
+	RecoveryConfigAsset = "v3.11.0/kube-apiserver/recovery-config.yaml"
+)
+
+var (
+	Scheme = runtime.NewScheme()
+	Codecs = serializer.NewCodecFactory(Scheme)
+)
+
+func init() {
+	metav1.AddToGroupVersion(Scheme, schema.GroupVersion{Version: "v1"})
+	utilruntime.Must(corev1.AddToScheme(Scheme))
+}
+
+type Apiserver struct {
+	KubeApiserverImage string
+	PodManifestDir     string
+	ResourceDirPath    string
+	RecoveryDirPath    string
```

Comment:
I know the other 3, but what is this?

Reply:
`RecoveryStatusPodResourceDir`  and everything is clear (as long as your monitor is big enough).

Reply:
the one used to store the recovery-apiserver resources

Reply:
just choose a better Go name. Doesn't help so much here in the comments ;-)